### DISCUSSION TITLE
New idea and devellopment possible

### DISCUSSION THREAD

**Discussion by MACTCD68** [PARADISE-AAV/rkd_package#13] [Ideas]
@Conorisco and @j2scott50 we can discuss new idea here and development of new function

**Comment by j2scott50** [comment]
Can you share the github project with me please, can't seem to find it. (GitHub is a little confusing for me as I haven't been able to invest much time in it at the minute)

**Reply by MACTCD68** [reply]
You just need to click on project when you are on this page.

**Comment by j2scott50** [comment]
While I like a discrete place to discuss unified goals, this limits the discussion to just us. We should invite all people on the basecamp here too.

**Reply by MACTCD68** [reply]
I will reshare the GitHub in Basecamp. After I can't force people to come.

**Reply by j2scott50** [reply]
Cool, maybe just highlight that the discussion of these transformed variables is in 'discussion' tab on github, same location as the project, so if people want to stay uptodate to link in here? (as it wasn't immediately obvious to me and therefore probably not to others either)

**Reply by kiplet** [reply]
Guys, you can progress technical discussions here, but please ensure that all high level ideas discussed yesterday are on basecamp. That is the PARADISE forum, not github

**Comment by Conorisco** [comment]
Early pipeline goal:

Initial output should be as an R markdown file with code (functions such as loadRKD) run in chunks, with some commentary.

**Reply by MACTCD68** [reply]
My main question now @Conorisco  for LoadRKD() is
Do we only load the data in a R object or do we load in a R object and save in a folder before transformation ?

**Comment by Conorisco** [comment]
I have created a new RMD that should act as the basis for the pipeline - setup, pathing and running the functions. Check it out!

I think this is a massive benefit as it allows other coders and easy method to get to the function they are working on without much fuss

**Reply by MACTCD68** [reply]
I change a bit the organisation with one folder with the function and a folder to test the function

**Comment by MACTCD68** [comment]
@kiplet Yagmur told me about a problem with RKD.ID. Do we need to solve it in the pipeline ?

**Reply by kiplet** [reply]
Yes. Please can you first review the patient.id variable to see if there are issues? The other option is to automatically introduce a transformation in rkd.id to account for the 16- IDs

**Reply by dogayy** [reply]
Some RKD ID s are starting with 16-.. so we have to replace them with Patient Id. 

Mark, after you  told me to use Patient Id for the merge, I checked and there are issues too; missing values. So instead of using patient id for merge, I prefer to deal with those rows in RKID and replace their value from Patiend Id column.

**Reply by MACTCD68** [reply]
@kiplet we have this strange patient with the patient.ID and we have only the patient.ID when Instrument =""
![image](https://user-images.githubusercontent.com/120493801/223152478-dcb54f46-f747-4b1f-92d0-c8a96a79742d.png)

**Reply by kiplet** [reply]
This is because they were also recruited to the STTAR bioresource. Perhaps best to see if we can transform the RKD.id if the 16- IDs are a problem

**Reply by MACTCD68** [reply]
Thanks @kiplet it is what we will do

**Comment by MACTCD68** [comment]
@kiplet I need to know the inclusion criteria to finalize the CleanRKD() function. 
Just want to be sure when I do the merge with the different data frame (Encounter, general characteristic) that I don't miss a variable that can be useful later

**Reply by kiplet** [reply]
Is Jen on this github? She is best placed to provide accurate definition. I will post on basecamp to ensure all in agreement

**Reply by MACTCD68** [reply]
@j2scott50

**Comment by MACTCD68** [comment]
@j2scott50 
When you tell this "([small_vessel_vas_immune] <> "1") AND ([small_vessel_vas_immune] <> "2") AND ([small_vessel_vas_immune] <> "3")", it mean ([small_vessel_vas_immune]=="") ?

**Comment by j2scott50** [comment]
Given EGPA is eliminated, this should probably only be = 1 or 2. Please refer to the code book for the specifics and match to the inclusion criteria on basecamp. The code book is accessible on redcap. If you don’t have access let me know and I will share. We should use this to double check the suggestions regardless of what I have written (human error possible).Kind regards,JenniferOn 9 Mar 2023, at 14:31, MACTCD68 ***@***.***> wrote:﻿
@j2scott50
When you tell this "([small_vessel_vas_immune] <> "1") AND ([small_vessel_vas_immune] <> "2") AND ([small_vessel_vas_immune] <> "3")", it mean ([small_vessel_vas_immune]=="") ?

**Reply by MACTCD68** [reply]
Ok @j2scott50 
The code book tell me that we have only 1,2,3 as option. 
![image](https://user-images.githubusercontent.com/120493801/224067744-f5c25e5d-96c7-453b-8e32-50f390fb1c37.png)
So ([small_vessel_vas_immune]=="") mean the same for me. But I prefer to be sure.

**Comment by j2scott50** [comment]
In that case yes, that should be the case. But I would do a check on the data using both approaches to make sure you obtain the same result. As you can see the data is very messy, so each time I apply a new rule I tend to run a quick test to make sure it has done what I thought it was going to doKind regards,JenniferOn 9 Mar 2023, at 15:14, MACTCD68 ***@***.***> wrote:﻿
Ok @j2scott50
The code book tell me that we have only 1,2,3 as option.

So ([small_vessel_vas_immune]=="") mean the same for me. But I prefer to be sure.
